docs/supported-tools.md

```diff
@@ -0,0 +1,271 @@
+in the configuration closure, or you can do so by adding a suppressions file:
+
+`pmd-rules.xml`
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  name="Novoda PMD rules"
+  xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+  xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+
+  ...
+
+  <!--region File exclusions-->
+  <exclude-pattern>.*\.kt</exclude-pattern>
+  <!--endregion-->
+
+  ...
+
+</ruleset>
+```
+
+## Findbugs
+[Findbugs](http://findbugs.sourceforge.net/) is a static analysis tool that looks for potential bugs in Java code. It does not support Kotlin.
+It can be used in both pure Java, and Android Java projects. It then only makes sense to have Findbugs enabled if you have Java code in your project.
+The plugin only runs Findbugs on projects that contain the Java or the Android plugin.
+
+### Configure Findbugs
+Enabling and configuring Findbugs for a project is done through the `findbugs` closure:
+
+```gradle
+findbugs {
+    toolVersion // A string, most likely '3.0.1' — the latest Findbugs release (for a long time)
+    exclude // A fileTree, such as project.fileTree('src/test/java') to exclude Java unit tests
+    excludeFilter // A file containing the Findbugs exclusions, e.g., teamPropsFile('static-analysis/findbugs-excludes.xml')
+    includeVariants { variant -> ... } // A closure to determine which variants (for Android) to include
+}
+```
+
+You can have multiple `exclude` statements.
+
+For more informations about Findbugs rules, refer to the [official website](http://findbugs.sourceforge.net/bugDescriptions.html).
+
+### Findbugs in mixed-language projects
+If your project mixes Java and Kotlin code, you most likely want to have an exclusion in place for all `*.kt` files. You can use the `exclude`
+in the configuration closure, or you can do so by adding a suppression to the filter file:
+
+`findbugs-excludes.xml`
+```xml
+<FindBugsFilter>
+
+  ...
+
+  <Match>
+    <Source name="~.*\.kt" />
+  </Match>
+
+  ...
+
+</FindBugsFilter>
+```
+
+## Detekt
+[Detekt](https://github.com/arturbosch/detekt) is a static analysis tool that looks for potential bugs and style violations in Kotlin code. It
+does not support Java. It can be used in both pure Kotlin, and Android Kotlin projects. It then only makes sense to have Detekt enabled if you
+have Kotlin code in your project. The plugin only runs Detekt on projects that contain the Kotlin or the Kotlin-Android plugin.
+
+### IMPORTANT: setup Detekt
+Unlike the other tools, the plugin **won't automatically add Detekt** to your project. If you forget to do it, the plugin will fail the build
+with an error like:
+
+```
+The detekt plugin is configured but not applied. Please apply the plugin in your build script.
+For more information see https://github.com/arturbosch/detekt.
+```
+
+In order to use Detekt in your project, you need to:
```

Comment:
What about having a link to official Detekt documentation? Users will definitely have to go there for details of the configuration anyways. So we would like them to go there and use their docs to learn how to add. And then use our docs to setup.

Reply:
There is several already, specifically pointing to the relevant bits, where applicable. Our instructions on how to add are different enough from the official ones that they are conflicting, so I chose not to add a direct reference here (but there are in the next sections)

Reply:
My goal was to keep this docs brief. Considering we will have more tools, having too much information may hurt. 
"How to add `detekt`" is out of our concern, imho. We expect it to be added, that's all.

Reply:
I think it would be a grave omission not to have this information here — after all you suggest yourself at line 222 to link to this documentation in the error message.

When we'll add more tools, we'll begin splitting this out into separate docs. I'm already planning that, but it doesn't really make much sense _right now_ imho. I will do it when I add Lint.

Reply:
After the comment from @mr-archano below asking for a split I put some more thought into the split. I will do it, before we get to Lint, but not in this PR because we'd lose the majority of the comments and discussions. A new PR will follow today with that change.